### PR TITLE
New Basemaps: Map Category - part 2 of 2

### PR DESCRIPTION
**Pull request by jenmerritt**
### Description

This PR is for the following sample categories:
- Map (part 2 of 2)

Notes specific to this PR: 
- Set initial map location: no mapping for National Geographic basemap so used best match at initial viewpoint and created new png. Also discussed with Trevor and agreed on this one to still follow the pattern of setting the viewpoint on the map view (and not the map itself)

### General comments on Basemap updates:

It updates `Map` and/or `Basemap` Constructors as per the new design. This may include the following changes depending on the sample:
- Replace `Basemap.Type` with `BasemapStyle`
- If a lat/long/zoom was being set in the `Map` constructor, this is replaced with a `Viewpoint` being set on the `MapView`.
- Any Readme Updates identified

In addition, the samples that include the above changes, or other services requiring an API Key going forward, have the API Key  set within the sample. This will then work in conjunction with the Gradle Task being added in PR 593.

Once 593 is approved and merged into `dev` we will merge those changes into these PRs and then update all the Samples to `100.10.0-xxxx` and do a round of testing of the Samples with all the API Key / Basemap changes at that time. (We have left them as 100.9.0 (in most cases) for now to minimize merge conflicts in the `build.gradle` file).

In the meantime, the updates listed above can be reviewed. Thanks!

### CODE REVIEW

File: map/manage-bookmarks/src/main/java/com/esri/samples/manage_bookmarks/ManageBookmarksSample.java [hunk 1]
```diff
@@ -117,9 +122,10 @@ public void start(Stage stage) {
       // add label and bookmarks to the control panel
       controlsVBox.getChildren().addAll(bookmarkLabel, bookmarkNames, addBookmarkButton);
 
-      ArcGISMap map = new ArcGISMap(Basemap.createImageryWithLabels());
+      // create a map with the imagery standard basemap style
```
Comment (JonLavi) [inline map/manage-bookmarks/src/main/java/com/esri/samples/manage_bookmarks/ManageBookmarksSample.java]: Same comment as in the other PR, might be worth checking if the plain english description is `standard imagery basemap`, as opposed to the technical enum name?

File: map/manage-operational-layers/src/main/java/com/esri/samples/manage_operational_layers/ManageOperationalLayersSample.java [hunk 2]
```diff
@@ -64,8 +65,12 @@ public void start(Stage stage) {
       stage.setScene(scene);
       stage.show();
 
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
       // create an ArcGISMap with the topographic basemap
```
Comment (JonLavi) [inline map/manage-operational-layers/src/main/java/com/esri/samples/manage_operational_layers/ManageOperationalLayersSample.java]: Just noticing most of the other comments for this line in other samples now say `create a map with..`

Might be nice to make it uniform?

So as you have in the next file:
```suggestion
      // create a map with the topographic basemap style
```

File: map/set-initial-map-location/README.md [hunk 3]
```diff
@@ -14,18 +14,20 @@ When the map loads, note the specific location and scale of the initial map view
 
 ## How it works
 
-1. Create a `Map`, specifying a basemap type, latitude and longitude in WGS84, and a level of detail.
+1. Create a `Map`, specifying a basemap style.
```
Comment (JonLavi) [inline map/set-initial-map-location/README.md]: This should either say `ArcGISMap` or 'map' without backticks. Easy to forget that `Map` == Key-value-Pair!

```suggestion
1. Create a `ArcGISMap`, specifying a basemap style.
```

Might be a good thing to create an issue to search for `Map` in backticks in all readme files and get rid of those relics once and for all ;)

**Review by JonLavi — APPROVED** [review]
Looks good @jenmerritt !
Just a couple of smaller pointers here too.
Approving so that you can move on once you made the necessary changes 👍

File: map/manage-bookmarks/src/main/java/com/esri/samples/manage_bookmarks/ManageBookmarksSample.java (hunk 1 above)
Comment (jenmerritt) [reply map/manage-bookmarks/src/main/java/com/esri/samples/manage_bookmarks/ManageBookmarksSample.java]: I'm just thinking on most of them I think it just says "with a basemap style". Maybe that is easier. Any thoughts @Rachael-E ?

File: map/manage-operational-layers/src/main/java/com/esri/samples/manage_operational_layers/ManageOperationalLayersSample.java (hunk 2 above)
Comment (jenmerritt) [reply map/manage-operational-layers/src/main/java/com/esri/samples/manage_operational_layers/ManageOperationalLayersSample.java]: Thanks yes that was the plan to make it consistent, so will tidy this one!

File: map/set-initial-map-location/README.md (hunk 3 above)
Comment (jenmerritt) [reply map/set-initial-map-location/README.md]: Ah cool, I will make a note 👍

Comment (jenmerritt) [reply map/set-initial-map-location/README.md]: Just did a search and it's actually only one other sample - so I'll update that in the relevant PR when I get back to it!

File: map/read-geopackage/src/main/java/com/esri/samples/read_geopackage/ReadGeoPackageSample.java [hunk 4]
```diff
@@ -52,11 +54,18 @@ public void start(Stage stage) {
...
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
       // create a map and add it to the map view
-      final ArcGISMap map = new ArcGISMap(Basemap.Type.STREETS, 39.7294, -104.8319, 11);
+      final ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_STREETS);
       mapView = new MapView();
       mapView.setMap(map);
 
+      // set a viewpoint on the map view
+      mapView.setViewpoint(new Viewpoint(39.7294, -104.8319, 300000.0));
```
Comment (Rachael-E) [inline map/read-geopackage/src/main/java/com/esri/samples/read_geopackage/ReadGeoPackageSample.java]: for LOD 11, we could go for 288895, rounded down to 288800 should be ok if you want it to look a little neater!

File: map/set-initial-map-location/src/main/java/com/esri/samples/set_initial_map_location/SetInitialMapLocationSample.java [hunk 5]
```diff
@@ -44,14 +46,20 @@ public void start(Stage stage) {
...
-      // create the MapView
+      // create a map with the navigation basemap style
+      ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_NAVIGATION);
+
+      // create the map view and set its map
       mapView = new MapView();
       mapView.setMap(map);
 
+      // set a viewpoint on the map view specifying latitude, longitude and scale
+      mapView.setViewpoint(new Viewpoint(56.075844, -2.681572, 700000));
```
Comment (Rachael-E) [inline map/set-initial-map-location/src/main/java/com/esri/samples/set_initial_map_location/SetInitialMapLocationSample.java]: LOD 10 would be 577790 scale here

**Review by Rachael-E — APPROVED** [review]
thanks @jenmerritt , just two instances of scale to consider again here.

**Comment by jenmerritt** [comment]
`dev` merged in (including API Key gradle tasks changes). Samples that have an API Key set have been tested running from the IDE.

Await merging these changes until release.

**Pull request merged**